tailscale_ssh_access/main.tf

```
@@ -0,0 +1,188 @@
+
+  settings = {
+    # `type=tailscale` sym_integrations have a required setting `api_token_secret`,
+    # which must point to a sym_secret referencing your Tailscale API Key
+    api_token_secret = sym_secret.tailscale_api_key.id
+  }
+}
+
+
+############ Tailscale Strategy Setup ##############
+
+# A target Tailscale group that your Sym Strategy can manage access to
+resource "sym_target" "tailscale_prod_group" {
+  type  = "tailscale_group"
+  name  = "main-prod-access"
+  label = "Prod SSH Access"
+
+  settings = {
+    # `type=tailscale_group` sym_targets have a required setting `group_name`,
+    # which must be the name of the Tailscale group the requester will be escalated to
+    # when this target is selected.
+
+    # The group name is defined in the "groups" section of your Tailscale Access Controls.
+    # Tailscale > Admin > Access Controls > Groups
+    group_name = "prod"
+  }
+}
+
+# A target Okta group that your Sym Strategy can manage access to
+resource "sym_target" "tailscale_staging_group" {
+  type  = "tailscale_group"
+  name  = "main-staging-access"
+  label = "Staging SSH Access"
+
+  settings = {
+    # `type=tailscale_group` sym_targets have a required setting `group_name`,
+    # which must be the name of the Tailscale group the requester will be escalated to
+    # when this target is selected.
+
+    # The group name is defined in the "groups" section of your Tailscale Access Controls.
+    # Tailscale > Admin > Access Controls > Groups
+    group_name = "staging"
+  }
+}
+
+# The Strategy your Flow uses to escalate to Okta Groups
+resource "sym_strategy" "tailscale" {
+  type           = "tailscale"
+  name           = "main-tailscale-strategy"
+  integration_id = sym_integration.tailscale.id
+
+  # This must be a list of `tailscale_group` sym_target that users can request to be escalated to
+  targets = [sym_target.tailscale_prod_group.id, sym_target.tailscale_staging_group.id]
+}
+
+############ Flow Setup ##############
+
+resource "sym_flow" "this" {
+  name  = "tailscale-ssh-access"
+  label = "Tailscale SSH Access"
+
+  template       = "sym:template:approval:1.0.0"
+  implementation = "${path.module}/impl.py"
+  environment_id = sym_environment.this.id
+
+  params = {
+    # By specifying a strategy, this Flow will now be able to manage access (escalate/de-escalate)
+    # to the targets specified in the `sym_strategy` resource.
+    strategy_id = sym_strategy.tailscale.id
+
+    # prompt_fields_json defines custom form fields for the Slack modal that
+    # requesters fill out to make their requests.
+    prompt_fields_json = jsonencode([
+      {
+        name     = "reason"
+        label    = "Why do you need access?"
+        type     = "string"
+        required = true
+      },
+      {
+        name           = "duration"
+        type           = "duration"
+        allowed_values = ["30m", "1h"]
+        required       = true
+      }
+    ])
+  }
+}
+
```

In my more recent examples, I've been sectioning this off with
```suggestion
############ Basic Environment Setup ##############
```